### PR TITLE
CI: SITL test run: select ROS version based on ROS_DISTRO

### PR DESCRIPTION
**Pull request by lamping7**
As we introduce SITL testing on Melodic this is important as we'll have Kinetic at the same time.

### CODE REVIEW

**Comment by lamping7** [comment]
The `ROS_DISTRO` env variable isn't available until we update the containers. 

Included here: https://github.com/PX4/containers/blob/f8a68d73f6e3e7db829e8a86fe9cc5976917ddbb/docker/px4-dev/Dockerfile_ros-kinetic#L9
and here:
https://github.com/PX4/containers/blob/f8a68d73f6e3e7db829e8a86fe9cc5976917ddbb/docker/px4-dev/Dockerfile_ros#L9

Changing to WIP...

**Comment by lamping7** [comment]
Working on updating images to support this --> https://github.com/PX4/Firmware/pull/11606

**Comment by lamping7** [comment]
rebased

File: test/rostest_px4_run.sh [hunk 1]
```diff
@@ -3,7 +3,7 @@
 DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
 PX4_SRC_DIR=${DIR}/..
 
-source /opt/ros/kinetic/setup.bash
+source /opt/ros/${ROS_DISTRO}/setup.bash
```
Comment (dagar) [inline test/rostest_px4_run.sh]: Default to kinetic if `ROS_DISTRO` doesn't exist?

Comment (lamping7) [reply test/rostest_px4_run.sh]: That's definitely a good idea.

Comment (lamping7) [reply test/rostest_px4_run.sh]: I didn't consider use outside of CI. Fixed.

Comment (TSC21) [reply test/rostest_px4_run.sh]: I don't think this is a good idea. What is the rationale for setting a default distro if we don't know if any is setup on the system. Also, if using ROS tools, the env variables have to always be set, which includes the ROS_DISTRO.

Comment (lamping7) [reply test/rostest_px4_run.sh]: This is a chicken and egg problem. `ROS_DISTRO` isn't set until we source `setup.bash`.

I see no problem setting a default version to correspond with the officially supported version that we use for testing. This is a testing script. Not a tools script.

**Comment by lamping7** [comment]
> Why not set the distro based on the OS distro?

This is a good idea, but I see issues with it. It would be fairly trivial if we assume this will only be used with Ubuntu. We could run something like `lsb_release -sc` and set it based on that. The problem is the assumption. To actually test for the users OS and set this appropriately, we need to consider Debian, Arch, and so on. I don't want to claim we're checking the distro and don't actually check for all the flavors compatible with Kinetic and Melodic respectively. So, I'd rather just fall back to the default ROS version supported in our CI system.

**Comment by TSC21** [comment]
Check this example: https://github.com/PX4/px4_ros_com/blob/master/scripts/setup_system.bash

**Comment by lamping7** [comment]
> Check this example: https://github.com/PX4/px4_ros_com/blob/master/scripts/setup_system.bash

That's exactly what I said we could do. I don't want to assume that everyone only uses Ubuntu.

**Comment by TSC21** [comment]
> > Check this example: https://github.com/PX4/px4_ros_com/blob/master/scripts/setup_system.bash
> 
> That's exactly what I said we could do. I don't want to assume that everyone only uses Ubuntu.

ROS officially support Ubuntu and Debian. Both Ubuntu and Debian support the `lsb-release` commands. The support for Arch, Gentoo, etc. for ROS is experimental, though it's also possible to verify the release version of them. In the end, I think what matters is what platforms we want to support based on where ROS can run (officially). Am I wrong to assume that?

**Comment by lamping7** [comment]
There's nothing wrong with your statement. It's just beyond the scope of what I was going for here. I was only trying to dynamically source the correct ROS setup for CI purposes as stated in the title and the original post. I still think the default matching what is officially used for testing is fine as it is only used for that purpose. If users want to run this on their own machine, they should have already sourced the correct setup script for their environment as defined in the installation instructions for ROS. If users run this inside the px4 docker container the environment variable will be set. I don't see any issues with this as is.

**Comment by dagar** [comment]
Good enough for now. We can continue to generalize further.

**Pull request merged**